### ISSUE TITLE
react-navi causing error in StrictMode

### ISSUE THREAD

**Issue by stramel** [frontarm/navi#72]
`SideEffect` is causing an error in `StrictMode`.

```js
SideEffect.prototype.componentWillMount = function componentWillMount() {
  mountedInstances.push(this);
  emitChange();
};
```

found this in `react-navi/dist/umd/react-navi.js`. Using `StrictMode` it causes this error:

```
Warning: Unsafe lifecycle methods were found within a strict-mode tree:
    in StrictMode (at App.tsx:7)
    in App (at src/index.tsx:6)

componentWillMount: Please update the following components to use componentDidMount instead: SideEffect(NullComponent)

Learn more about this warning here:
https://fb.me/react-strict-mode-warnings
```

**Comment by jamesknelson** [comment]
This looks to be caused by the dependency on react-helmet. I'm going to need to update Navi to either use react-helmet-async, or just remove react-helmet and leave head/title management to the application.

Would love to hear any thoughts on the best option here.

**Comment by stramel** [comment]
@jamesknelson Ah, that makes sense as to why I couldn't find it in this codebase. 

My preference would be to remove the embedded version of `react-helmet` and update the docs a bit to show how they can easily be used together. It should drop the bundle size for `react-navi` for those not using it, however, it would be a breaking change.

**Comment by jamesknelson** [comment]
I think I'm leaning that way too, although you're right that it'd be a breaking change.

The next release should provide a way to add custom matchers like `withHead` and `withTitle`, so perhaps what I'll do is extract withHead/withTitle to a separate package in the next release, and switch that package to use react-helmet-async at the same time.

In the meantime, if react-helmet-async is API compatible with react-helmet, I would accept a PR to switch over in a minor release to help with `<StrictMode>`.

**Comment by stramel** [comment]
@jamesknelson Unfortunately, I don't believe that would be a non-breaking change either as `react-helmet-async` requires `<Helmet>` be wrapped with their `Provider`.

> <Helmet> usage is synonymous, but server and client now requires <HelmetProvider> to encapsulate state per request.

While I think it would be smart to switch to using it for the examples or with the separate package you mentioned releasing in the future.

> The next release should provide a way to add custom matchers like withHead and withTitle, so perhaps what I'll do is extract withHead/withTitle to a separate package in the next release, and switch that package to use react-helmet-async at the same time.

I don't believe it would be a non-breaking change 😞

**Comment by jamesknelson** [comment]
Just released 0.13.0-alpha.0 which fixes this. Will publish a 0.13.0 as soon as docs are done.

I've split the helmet integration code out into separate packages: `react-navi-helmet` and `react-navi-helmet-async`. They both export a `<HelmetProvider>` component which you can import and wrap your app with to integrate Helmet with Navi if you want it -- or you can just leave them and manage head/title without Navi.

 ```jsx
    import HelmetProvider from 'react-navi-helmet-async'

    ReactDOM.render(
      <HelmetProvider>
        <Router routes={routes}>
      </HelmetProvider>,
      document.getElementById('root')
    )
```
